### ISSUE TITLE
movie: model must allow for reserved, checked out, or available state

### ISSUE THREAD

**Issue by dlukeomalley** [dormbase/dormbase#20]
Currently just uses a boolean for available, but residents can both reserve a film and then go to desk to check it out. The whole model needs to be reworked regardless, so keep this in mind when touching it.

_NOTE_ This entire system needs to be reworked to also support games. We must add a DVD type (ie, BlueRay, etc).

**Comment by achernya** [comment]
I think we should have a separate movie-status table that has a full log of all requests, so that we don't ever write to the movies. That way we can easily support multiple copies without polluting the database or making complex queries.

**Comment by ievans** [comment]
Agreed; we should note however that there may be cases where the movie is "unavailable" in the sense that the disc is missing/defective and it will be replaced soon, but we don't want to pull it from the db. So I think the availability field is still valid in some context.

**Comment by achernya** [comment]
I disagree, the availability should be tracked at the disk-level, which is different from the description level.

**Comment by ievans** [comment]
No, I'm just saying "available" still needs to be a field in the meta-table. Does not belong in the Movies model though.
